v3/lints/cabf_br/lint_dnsname_contains_prohibited_reserved_label_test.go

```diff
@@ -0,0 +1,40 @@
+package cabf_br
+
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"testing"
+
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/test"
+)
+
+func TestDNSNameHasProhibitedReservedLabel(t *testing.T) {
+	inputPath := "dnsNameProhibitedReservedLabel.pem"
+	expected := lint.Error
+	out := test.TestLint("e_dnsname_contains_prohibited_reserved_label", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
+
+func TestDNSNameHasXNLabel(t *testing.T) {
+	inputPath := "dnsNameXNLabel.pem"
+	expected := lint.Pass
+	out := test.TestLint("e_dnsname_contains_prohibited_reserved_label", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
```

Comment:
More of a nit, but just noting we'd been doing a lot more table-based tests in the new lints ( e.g. https://github.com/zmap/zlint/blob/d5ab97e9ca1cef2c7b594672c0265cef3b703637/v3/lints/cabf_br/lint_san_dns_name_onion_invalid_test.go )

One of the reasons I ask here is the use of the RegEx makes me wonder how Golang would handle smuggled UTF-8 - would the Regex match the character or the rune? For example, if a string was `"{two byte character, -, -}"`, would RegEx thinks it fails (because the two byte UTF-8 character is only one rune), or would it match? This is a _very_ contrived example, because obviously `IA5String` is not supposed to convey these characters, and we'd hopefully get a decode failure far sooner, but uh, you know how messy it is with ASN.1 string types. I _believe_ the example alternative I gave might handle this better, but I'm also not sure.

That suggests adding another test might be useful, which would be a lot of copy/paste boiler plate, which is where the table approach comes in :)

Reply:
Looks like Go `regexp` package is Unicode-aware: https://play.golang.org/p/x1-e_ZqwM2S. I think using Regex is fine in this case given the `IA5String` encoding of `dNSNames` and this code will work unchanged if we apply to subject CNs, which might be (but hopefully not) multi-byte.

Given that there's currently only two test cases, I went with the current approach, which is in line with the `CONTRIBUTING.md` document guidance. Happy to switch to table-based, but the universal switch to table-based tests should probably be documented in that doc so new lint contributors do the right thing the first time.